python/metatensor-core/tests/serialization.py

```diff
@@ -193,6 +194,28 @@ def test_save_warning_errors(tmpdir, tensor):
             metatensor.save(tmpfile, tensor.block(0).values)
 
 
+def test_save_pathlib(tmpdir, tensor):
+    # does not have .npz ending and causes warning
+    tmpfile = Path("serialize-test")
+
+    with pytest.warns() as record:
+        with tmpdir.as_cwd():
+            metatensor.save(tmpfile, tensor)
+
+    expected = f"adding '.npz' extension, the file will be saved at '{tmpfile}.npz'"
+    assert str(record[0].message) == expected
```

Comment:
You can use the same syntax for `pytest.raises` and `pytest.warns`

```suggestion
    expected = f"adding '.npz' extension, the file will be saved at '{tmpfile}.npz'"
    with tmpdir.as_cwd():
        with pytest.warns(UserWarning, match=expected):
            metatensor.save(tmpfile, tensor)
```